### DISCUSSION TITLE
Rating maps / "User rating"

### DISCUSSION THREAD

**Discussion by ArijanJ** [ppy/osu#29790] [Ideas]
This feature only exists in stable for now, available by scrolling down after a map is completed.

![image](https://github.com/user-attachments/assets/8605ed9c-3691-41fa-95a6-37f31e6e1bb6)

Rather than a regular feature request, I'm curious whether we would be willing to convert this system to a two (or three)-choice "Did you like this map?" prompt/notification/toast, as that is what most apps use currently.

There is some writing on this topic (https://kirkville.com/why-youtube-switched-from-5-star-ratings-to-thumbs-updown-in-2009-daring-fireball/), but it should give a more accurate and less mentally taxing way to rate maps that you enjoyed.

I personally prefer this system, and I find that sorting by favorites instead of by rating actually gives higher quality maps (not sure how well these two compare), but it would be nice to have a way of showing similar support without having to fully favorite the map.

If not, we could just put the 10 stars in lazer as well, no big deal - but I'm not sure of the implementation, couldn't find anything in the API docs :(

**Comment by ArijanJ** [comment]
Forgot to mention, #13015 is about the exact same thing but it was 5 years ago so opening a new one.

**Comment by Alib234** [comment]
I think for now maybe the 10 star rating system should be implemented and then later an overhaul of the system for 2/3 rating options, although that's not for me to decide.

Making this a notification doesn't sound good to me as it can be easily missed, possibly filling notification feed and may annoy some people.

It should probably be in the results screen after you finish a map similar to how stable does it?

![rate this map mockup](https://github.com/user-attachments/assets/3353a8b2-3480-457c-94dd-b15586341104)

**Reply by ArijanJ** [reply]
I could recommend a dismissable notification so that it doesn't clutter up score screenshots, but yes it should only be available at the score card screen just like on stable:

![image](https://github.com/user-attachments/assets/f6265e8b-e643-408b-9c90-9843d9631eab)

But since we're most likely going to end up making it be stars, I'm not really sure where a good place to put it would be.

**Reply by Alib234** [reply]
also your screenshot made me realize my idea only would work if people used .8x UI scaling like I do, I forgot to check if there is space there on the gigantic default scaling

**Reply by Tenexxt** [reply]
Since there is a result screen rework, I'd recommend checking it out and asking if there is any sort of space or add this in instead of trying to make this into a notification or something similar.
